configuration/config2/Status.cpp

```diff
@@ -0,0 +1,73 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2017 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "Status.hpp"
+
+
+void Status::addStatusMsg(enum statusMsg::msgLevel level, const std::string &nodeId, const std::string &name, const std::string &referNodeId, const std::string &msg)
```

Comment:
This status class has a generic name, but it seems to be specific to validate() actions (at least this method makes it seem specific to actions involving nodeIds).
 If that's the case let's label it as such "ValidateStatus"
otherwise, let's provide a truly generic addStatusMsg (enum statusMsg::msgLevel level,  const std::string &msg) as well.

Reply:
I changed the method per our conversation and provided overloads that eliminate the unneeded parameters.